### PR TITLE
improve localhost detection

### PR DESCRIPTION
**Pull request by James-Mart**

### CODE REVIEW

File: services/user/AdminSys/ui/src/pages/dashboard-page.tsx [hunk 1]
```diff
@@ -1,6 +1,11 @@
-const DASHBOARDS_PREFIX =
-    window.location.hostname === "localhost"
-        ? "http://localhost:8091/grafana"
+function isLocalhost() {
+    const hostname = window.location.hostname;
+    return hostname.includes('localhost') || hostname.includes('127.0.0.1');
```
Comment (swatanabe) [inline services/user/AdminSys/ui/src/pages/dashboard-page.tsx:3]: I would probably restrict the match to require any leading or trailing components to be separated by a `.`.  Also, one more way to spell localhost is: `[::1]`

File: services/user/AdminSys/ui/src/pages/dashboard-page.tsx [hunk 2]
```diff
@@ -1,6 +1,18 @@
-const DASHBOARDS_PREFIX =
-    window.location.hostname === "localhost"
-        ? "http://localhost:8091/grafana"
+function isLocalhost() {
+    const hostname = window.location.hostname;
+    const localhostPatterns = [
+        /^(.*\.)?localhost:.*$/,
+        /^(.*\.)?127\.0\.0\.1(\..*)?$/,
```
Comment (swatanabe) [inline services/user/AdminSys/ui/src/pages/dashboard-page.tsx:5]: This misses `127.0.0.1:8080`

File: services/user/AdminSys/ui/src/pages/dashboard-page.tsx [hunk 3]
```diff
@@ -1,6 +1,18 @@
-const DASHBOARDS_PREFIX =
-    window.location.hostname === "localhost"
-        ? "http://localhost:8091/grafana"
+function isLocalhost() {
+    const hostname = window.location.hostname;
+    const localhostPatterns = [
+        /^(.*\.)?localhost:.*$/,
```
Comment (swatanabe) [inline services/user/AdminSys/ui/src/pages/dashboard-page.tsx:4]: If `psinode` is running on port `80` or `443`, the hostname might not contain a port.

**Comment by James-Mart** [comment]
Regexes should now cover:
    localhost
    localhost:port
    page.localhost
    page.localhost:port

    127.0.0.1
    127.0.0.1:port
    page.127.0.0.1.sslip.io
    page.127.0.0.1.sslip.io:port

    [::1]
    [::1]:port
    page.--1.sslip.io
    page.--1.sslip.io:port

**Review by swatanabe — APPROVED** [review]
I was going to complain that this matches `localhost42`, but a quick look at my `/etc/hosts` shows that that may be a good thing:
```
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
```

**Pull request merged**